src/libYARP_os/src/yarp/os/LogStream.h

```diff
@@ -247,26 +249,80 @@ class YARP_os_API LogStream
         return *this;
     }
 
-    template <typename T>
-    inline LogStream& operator<<(const std::vector<T>& t)
+#ifndef SWIG_PREPROCESSOR_SHOULD_SKIP_THIS
+    template <typename T1, typename T2>
+    inline LogStream& operator<<(const std::pair<T1, T2>& t)
+    {
+        stream->oss << '{' << t.first << ", " << t.second << '}';
```

Comment:
Perhaps I should have split this chain into `stream->oss << '{'; *this << t.first; ...`. I'll check it.

Reply:
Yes, this needs a similar treatment as vectors in order to be able to use nested containers (e.g. `std::pair<int, std::vector<double>>`) and avoid excess space chars. Done.